app/Http/Controllers/SessionController.php

```diff
@@ -73,9 +73,9 @@ public function init(): array
 				$user = Auth::user();
 				$return['user'] = $user?->toArray();
 				$return['rights'] = [
-					'is_admin' => Gate::check(UserPolicy::IS_ADMIN),
-					'is_locked' => $user?->is_locked ?? true,
-					'may_upload' => $user?->may_upload ?? false,
+					'is_admin' => Gate::check(UserPolicy::IS_ADMIN, User::class),
+					'is_locked' => !Gate::check(UserPolicy::CAN_EDIT_SETTINGS, User::class), // the use of the negation should be removed later
+					'may_upload' => Gate::check(UserPolicy::CAN_UPLOAD, User::class),
```

Comment:
Actually, this doesn't make much sense.

`CAN_UPLOAD` is not a right on a user object (because one does not upload users), but rather a right on photos as well as albums.

Or in terms of https://github.com/LycheeOrg/Lychee/issues/1481#issuecomment-1229542549:

`may_upload` is a capability of the user and should stay there, but not be defined as the outcome of a policy. This capability contributes to rights like "may create album" or "may add photos". Only the latter are determined by a policy.

However, I also concede that the gate checks `UserPolicy::CAN_UPLOAD`, etc., have already been defined previously are are not introduced in by this PR. In other words, the actual fault has already happened earlier.

Hence, for the time being, I am fine with using this work-around here as a patch.

Reply:
https://github.com/LycheeOrg/Lychee/pull/1511 :) This aims to fix that.